### PR TITLE
Show total length of songs after bust

### LINKED ISSUE #69: Add total sum of lengths of all played songs to "Thanks for coming"
We wouldn't want to let people know how long the stream is going to be from the start, because then it lets you guess the lengths of the last songs. However it's cool to know much content was created overall at the end.

### PR DESCRIPTION
**Pull request by Cephian**
Closes #69.

Sums the total lengths of all submissions to show at the end. This branch was tested in production at Busty's 6 (and it worked!)
image

The actual calculation is occurs when `!list` is called, but this doesn't increase list command latency since it is calculated after messages are sent and pinned.

### CODE REVIEW

**Comment by Cephian** [comment]
I'm not totally sure why GitHub still says this branch has a merge conflict, but the "conflict" would be very easy to solve manually in the GitHub interface when merging, so consider this ready to merge.

**Comment by anoadragon453** [comment]
@Cephian I fixed the conflict by doing:

```
git checkout total_len
git pull origin main
<fix conflicts>
git add -u && git commit
```

File: main.py [hunk 1]
```diff
@@ -255,6 +257,21 @@ def song_format(
...
+        audio = MutagenFile(filename)
+        if audio:
+            return audio.info.length
+    except MutagenError as e:
+        # Ignore file and move on
+        print(f"Error reading length of {filename}:", e)
+        return None
+    except Exception as e:
+        print(f"Unknown error reading length of {filename}:", e)
+        return None
```
Comment (anoadragon453) [inline main.py:272]: `return None` could be extracted from both exception cases.

File: main.py [hunk 2]
```diff
@@ -65,6 +65,8 @@
 original_bot_nickname: Optional[str] = None
 # Allow only one async routine to calculate !list at a time
 list_task_control_lock = asyncio.Lock()
+# Total length of all songs
```
Comment (anoadragon453) [inline main.py:68]:
```suggestion
# Total length of all songs in seconds
```

File: main.py [hunk 3]
```diff
@@ -417,7 +451,10 @@ async def play_next_song(skip_count: int = 0) -> None:
 
         # Say our goodbyes
         embed_title = "❤️‍🔥 That's it everyone ❤️‍🔥"
-        embed_content = "Hope ya had a good **BUST!**"
+        embed_content = "*Total length of all submissions: {}*\n".format(
+            format_time(int(total_song_len))
+        )
```
Comment (anoadragon453) [inline main.py:456]: What do you think about putting the "total length" text at the bottom of the end message instead?

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
A few small changes, but overall looks pretty good!

File: main.py [hunk 4]
```diff
@@ -255,6 +257,21 @@ def song_format(
     return content
 
 
+# Get length of a song
+def get_length(filename: str) -> Optional[float]:
+    try:
+        audio = MutagenFile(filename)
```
[the remark this answers is not in the document]
Comment (Cephian) [reply main.py:263]: If we do this it should be in a separate PR. I'd also want to see if it's doing things like storing the entire tag data (including all cover images) in the object which would increase our memory usage alot. At the very least, we are definitely making separate mutagen calls for both length and tag data when `!list`ing, and separate calls for both cover art and tag data when playing. which could probably be refactored and combined.

**Review by anoadragon453 — APPROVED** [review]
lgtm!

**Pull request merged**